python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -181,9 +209,53 @@ def latest_time_window_slice(self) -> "AssetSlice":
                 else self._asset_graph_view.create_empty_slice(self.asset_key)
             )
 
-        # Need to handle dynamic and multi-dimensional partitioning
+        if isinstance(self._partitions_def, MultiPartitionsDefinition):
+            if not self._partitions_def.has_time_window_dimension:
+                return self
+
+            multi_dim_info = self._get_multi_dim_info()
+            last_tw = multi_dim_info.tw_partition_def.get_last_partition_window(
+                self._asset_graph_view.effective_dt
+            )
```

Comment:
Since this is going to become a core part of the system, IMO we should put the necessary time window dimension accessors on `MultiPartitionsDefinition` instead of having this one-off `MultiDimInfo` class.

EDIT: Looks like you did actually add to `MultiPartitionsDefinition`?

Reply:
I just added some helper methods to `MultiPartitionsDefinition`. I think we should add to the core layer in a follow up, alongside an effort to represent multi-partioning in the underlying `PartitionsSubset` more efficiently.